joblib/test/test_hashing.py

```diff
@@ -408,30 +412,24 @@ def test_hashes_stay_the_same_with_numpy_objects():
                 rng.randint(10, size=20).astype('<i8'),
                 rng.randn(10).astype('<f4')
             ]
-        }
+        },
+        non_picklable
     ]
 
     # These expected results have been generated with joblib 0.9.0
     expected_dict = {'py2': ['80f2387e7752abbda2658aafed49e086',
                              '0d700f7f25ea670fd305e4cd93b0e8cd',
                              '83a2bdf843e79e4b3e26521db73088b9',
-                             '63e0efd43c0a9ad92a07e8ce04338dd3'],
+                             '63e0efd43c0a9ad92a07e8ce04338dd3',
+                             '140de081a68afd241ea774d014130a7f'],
```

Comment:
I think you should add tests for non-picklable objects in a separate test_function. As the comment is saying, these hashes have been generated with joblib 0.9.0 (full disclosure actually 0.9.2) as a reference.